src/cli.rs

```diff
@@ -159,6 +159,27 @@ pub struct RootOpts {
     )]
     pub internal_log_rate_limit: u64,
 
+    /// Set the duration in seconds to wait for graceful shutdown after SIGINT or SIGTERM are received.
+    /// After the duration has passed, Vector will force shutdown.
+    #[arg(
+        long,
+        default_value = "60",
+        env = "VECTOR_GRACEFUL_SHUTDOWN_DURATION",
+        group = "graceful-shutdown-duration"
+    )]
+    pub graceful_shutdown_duration: NonZeroU64,
+
+    /// Never time out while waiting for graceful shutdown after SIGINT or SIGTERM received. This is useful
+    /// when you would like for Vector to attempt to send data until terminated by a SIGKILL. Overrides/cannot
+    /// be set with --graceful-shutdown-duration.
+    #[arg(
+        long,
+        default_value = "false",
+        env = "VECTOR_NO_FORCED_SHUTDOWN",
+        group = "graceful-shutdown-duration"
+    )]
+    pub no_forced_shutdown: bool,
```

Comment:
I think having this option be more similarly named to the `graceful_shutdown_secs` option would make it easier to find. I'd suggest `no_graceful_shutdown`.

I raised a question of whether we want to establish this configuration flag pattern for options that are also disable-able internally, but, assuming we do, I'd suggest we adopt the general pattern of adding a `no_` prefix and removing any units suffix.

Reply:
That's a good point 😄 I like those new names you suggested.

Reply:
We discussed and feel good about the current UX 👍